backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceInDatabase.java

```diff
@@ -0,0 +1,332 @@
+          }
+        });
+    return super.submit(task);
+  }
+
+  @Override
+  public Task getTask(String id) {
+    Task task = super.getTask(id);
+    if (task == null) {
+      return getTaskFromDatabase(id);
+    }
+    return task;
+  }
+
+  public Task getTaskFromDatabase(String id) {
+    final StringBuilder json = new StringBuilder();
+    database.tx(
+        db -> {
+          db.becomeAdmin();
+          List<Row> rows =
+              db.getSchema(systemSchemaName)
+                  .getTable("Jobs")
+                  .where(f("id", EQUALS, id))
+                  .retrieveRows();
+          if (rows.size() > 0) {
+            Row jobRow = rows.get(0);
+            json.append(jobRow.getString("log"));
+          }
+        });
+    try {
+      return (new ObjectMapper().readValue(json.toString(), Task.class));
+    } catch (Exception e) {
+      throw new MolgenisException("getTask(" + id + ") failed", e);
+    }
+  }
+
+  @Override
+  public String submitTaskFromName(
+      String scriptName, final String userName, final String parameters) {
+    StringBuilder result = new StringBuilder();
+    database.tx(
+        db -> {
+          db.becomeAdmin();
+          Schema systemSchema = db.getSchema(this.systemSchemaName);
+          // retrieve the script from database
+          List<Row> rows =
+              systemSchema.getTable("Scripts").where(f("name", EQUALS, scriptName)).retrieveRows();
+          if (rows.size() != 1) {
+            throw new MolgenisException("Script " + scriptName + " not found");
+          }
+          Row scriptMetadata = rows.get(0);
+          String user = userName;
+          if (user == null) {
+            user = scriptMetadata.getString("cronUser");
+            if (user == null) {
+              user = systemSchema.getDatabase().getActiveUser();
+            }
+          }
+          db.setActiveUser(user);
+          if (scriptMetadata != null) {
+            if (scriptMetadata.getBoolean("disable") != null
+                && scriptMetadata.getBoolean("disable")) {
+              throw new MolgenisException("Script " + scriptName + " is disabled");
+            }
+            // submit the script
+            result.append(
+                this.submit(
+                    new ScriptTask(scriptMetadata)
+                        .parameters(parameters)
+                        .token(
+                            JWTgenerator.createTemporaryToken(
+                                systemSchema.getDatabase(),
+                                systemSchema.getDatabase().getActiveUser()))
+                        .submitUser(user)));
+          } else {
+            throw new MolgenisException("Script execution failed: " + scriptName + " not found");
+          }
+        });
+    return result.toString();
+  }
+
+  private void save(Task task) {
+    this.saveWithOutput(task, null);
+  }
+
+  private void saveWithOutput(Task task, File outputFile) {
+    Row jobRow =
+        row(
```

Comment:
Row.fromTask(task)

Reply:
?

Reply:
Ah, I think there are some differences between this and the fromTask